pydbtools/_wrangler.py

```diff
@@ -506,3 +507,36 @@ def save_query_to_parquet(sql: str, file_path: str) -> None:
     df.to_parquet(file_path)
 
     return None
+
+
+def s3_path_join(base: str, url: str, allow_fragments=True) -> str:
+    """
+    Joins a base S3 path and a URL. Acts the same as urllib.parse.urljoin,
+    which doesn't work for S3 paths.
+
+    Args:
+        base (str): Base S3 URL
+        url (str):
+    """
+    p = urlparse(base)
+    return urlunparse(p._replace(path=urljoin(p.path, url, allow_fragments)))
+
+
+def dataframe_to_temp_table(df: pd.DataFrame, table: str) -> None:
+    """
+    Creates a temporary table from a dataframe.
+
+    Args:
+        df (pandas.DataFrame): A pandas DataFrame
+        table (str): The name of the table in the temporary database
+    """
+    user_id, table_dir = get_user_id_and_table_dir()
+    db = get_database_name_from_userid(user_id)
+    _create_temp_database(db)
+    wr.s3.to_parquet(
+        df,
+        path=s3_path_join(table_dir, f"{table}.parquet"),
```

Comment:
You can use `os.path.join` for joining s3 paths
also you just need a folder path here, so `os.path.join(table_dir, table)` rather than a file name

Reply:
`os.path.join` won't work on Windows - we probably don't have any Windows users but it may be better to keep it portable. For the folder path - [db1d8ea](https://github.com/moj-analytical-services/pydbtools/pull/70/commits/db1d8eafa721908dd0b0608b7926d098ebe898cd)

Reply:
`os.path.join` won't work for Windows users - I don't think we have any at the moment but just in case. Removing the `.parquet` suffix gives

```
 InvalidArgumentValue: The specified path: s3://blah/sales, does not match the existing Glue catalog table path: s3://blah/sales.parquet
```

Reply:
Curious! It must be baked into the code somewheere else - either way, it's fine as is. Resolving convo